### ISSUE TITLE
Change in behavior between pylance 2021.12.1 and 2022.2.0 on a pandas example

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#2985]
**Describe the bug**
Using the pandas stubs shipped with pylance 2021.12.1 on pylance 2022.2.0 reports new errors

**To Reproduce**
Using code below, do the following:
Install pylance 2021.12.1 and no errors are reported
With pylance 2022.2.0 and you get errors. 

Then to see it is not a problem with the stubs, do the following:

Take the pandas stubs from `~/.vscode/extensions/ms-python.vscode-pylance-2021.12.1/dist/bundled/stubs/pandas` and use them in the directory  `~/.vscode/extensions/ms-python.vscode-pylance-2021.22.0/dist/bundled/stubs/pandas`

Then you get the error using pylance 2022.2.0 using the stubs from 2021.12.1 .

Error is:  
```
Cannot access member "values" for type "Series[Dtype@__getitem__]"
```

Note a similar error is achieved if you try to access `s2.name` or `s2.hasnans` but not `s2.items()`, so it has something to do with looking up the type of a property.

**Expected behavior**
No error

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import pandas as pd

df = pd.DataFrame({"x": [1, 2, 3], "y": [5, 6, 7]})

d2 = df[["x"]]

s2 = d2.loc[0]

sv = s2.values

```

**VS Code extension or command-line**
VSCode versions as described above

**Additional context**
Same issue seems to be true with pylance 2022.1.3, but things are OK with pylance 2022.1.1

The methodology above proves the issue is with pyright, not with the stubs, unless there is something in the stubs you are shipping that is incorrect and then you have to resolve that with the owners of those stubs.

**Comment by erictraut** [comment]
I've isolated the commit that changed this behavior: https://github.com/microsoft/pyright/commit/de699f8bd416d2ad029cc72f511b201e158731f7.

Part of the problem here is that the pandas stub file `core/frame.pyi` is using TypeVars incorrectly in some of the overloaded signatures for `_LocIndexerFrame.__getitem__`. It's using the TypeVar `Dtype` in the return type annotation without including it in any input parameter type annotations. That means there's no way for the TypeVar constraint solver to determine the value of `Dtype` when `__getitem__` is invoked, and that leaves the TypeVar "unsolved", which creates downstream problems. Furthermore, `Dtype` is a constrained TypeVar, but its constraints are not compatible with the constraints for the type parameter supported by `Series`, so the use of `Series[Dtype]` is an error. The return types for the `__getitem__` method overloads should be changed to `Series[Any]` or `Series[S1]` where `S1` is the TypeVar used in the declaration of the `Series` class.

Because of this bug in the stubs, the `__getattr__` method cannot be bound because its `self` parameter has the type `Series[S1]`, and `Series[Dtype]` is incompatible.

I'll investigate further to see if there's a way we can mitigate the impact of this change in pyright, but we really should fix the stubs because they're the real source of the problem here. (cc @gramster)

**Comment by Dr-Irv** [comment]
> Because of this bug in the stubs, the `__getattr__` method cannot be bound because its `self` parameter has the type `Series[S1]`, and `Series[Dtype]` is incompatible.

Thanks.  For our use, I changed __some__ of the `__getitem__` returns in my copy of the stubs so that we don't get the error any more, but `Series[Dtype]` is used in many places in these files:
```
./core/frame.pyi
./core/generic.pyi
./core/groupby/generic.pyi
./core/reshape/concat.pyi
./core/series.pyi
```
and I think @gramster may need to look at that as opposed to me doing a PR. 

I opened up an issue here to track it:  https://github.com/microsoft/python-type-stubs/issues/144

**Comment by erictraut** [comment]
I've confirmed that the behavior change is due to a bug fix. The current behavior is correct, and the old behavior was not. It was masking a legitimate type error.

I'm looking at ways to work around this bug in the stubs, but all of the mitigations I've come up with so far would result in regressions from a type checking perspective — or would carry risk of introducing a change in behavior that could negatively impact existing pyright and pylance users who have type checking enabled.

IMO, the best fix is to address the underlying problem in the pandas type stubs that are bundled with pylance.

In `pandas/_typing.pyi`, the `Dtype` TypeVar is defined as follows:
```python
Dtype = TypeVar("Dtype", bool, int, float, object)
```
Because it includes `object` as a constraint, it is not compatible with the TypeVar `S1`, which is declared in the same file as follows:
```python
S1 = TypeVar("S1", str, bytes, datetime.date, datetime.datetime, datetime.timedelta, bool, int, float, complex)
```

Should `Dtype` be the same as `S1`? It's also OK if `Dtype` includes a subset of the constraints in `S1`. But it's not OK for it to contain a disjoint set, hence the problem with `object`.

@gramster, if you're OK making that change, then I would prefer to close this bug without taking any further action in terms of changing pyright's type checking behaviors.
